core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
I like the idea of reusable validation rules. We can define a bunch of predefined rules, so users can use them out of the box. The suggested syntax looks good to me. We also may consider the following syntax:
```scala
RequiredIntParam("a") which (ShouldBePositive and ShouldBeLessThen(100))
```
In case if we have method `and` defined on a validation rule.